From a333eb7af233fc2df0fcce2d4accf469a01f8980 Mon Sep 17 00:00:00 2001
From: Zhikui Ren <zhikui.ren@intel.com>
Date: Tue, 27 Feb 2024 14:32:13 -0800
Subject: [PATCH] Avoid unnecessary restart

When tryRestartControlLoops is call because new configuration is
detected, asio::post is used to put work in the queue immediately.
There is no need to keep any delayed restart work.
Cancel the active timer to avoid unnecessary restart.

Tested:
I made restartControlLoops call slow and power cycle the host and
verified that extra restart did not happen with the code change.

Change-Id: I4d131651b6b997f24f4ee057d7b20e3a6073e351
Signed-off-by: Zhikui Ren <zhikui.ren@intel.com>
---
 main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/main.cpp b/main.cpp
index fc09d8b..f99fb5e 100644
--- a/main.cpp
+++ b/main.cpp
@@ -194,6 +194,9 @@ void tryRestartControlLoops(bool first)
     // first time of trying to restart the control loop without a delay
     if (first)
     {
+        // first can be called because new config is detected
+        // while waiting for the retry timer
+        timer.cancel();
         boost::asio::post(io,
                           std::bind(restartLbd, boost::system::error_code()));
     }
-- 
2.25.1

